src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,31 +42,67 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        // Split address string to parts
+        String[] addressParts = address.split(ADDRESS_PARTS_DELIMITER);
+        _block = new Block(addressParts[ADDRESS_PARTS_INDEX_BLOCK].trim());
+        _street = new Street(addressParts[ADDRESS_PARTS_INDEX_STREET].trim());
+        _unit = new Unit(addressParts[ADDRESS_PARTS_INDEX_UNIT].trim());
+        _postalCode = new PostalCode(addressParts[ADDRESS_PARTS_INDEX_POSTALCODE].trim());
```

Comment:
While our coding standard allows this naming format (i.e. prefix local variables with underscore), you should be consistent with rest of the code which uses a different naming convention.